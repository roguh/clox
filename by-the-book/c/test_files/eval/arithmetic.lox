print(1 + 1);
print(2 - 1);
print(2 * 2);
print(4 / 2);
print(1 & 1);
print(1 | 1);
print(1 ^ 1);
print(~1);
print(1 << 12);
print(38 >> 1);
print(2 >> 1);
print(5 % 2);

// Test hardcoded series for Napier's constant as scaled integer
print(100000000 / 1 + 200000000 / 2 + 300000000 / 6 + 400000000 / 24 + 500000000 / 120 + 600000000 / 720 + 700000000 / 5040 + 800000000 / 40320);

print(1.0 / (113.0 / 355.0));

print(5.0 % 2);
print(3.14159265 % 1.5);
print(2 ** 3);

// Test golden ratio
print(0.5 + 5 ** 0.5 * 0.5);

// Test hardcoded series for Napier's constant
print(1 / 1 + 2 / 2 + 3.0 / 6 + 4.0 / 24 + 5.0 / 120 + 6.0 / 720 + 7.0 / 5040 + 8.0 / 40320);

// Test hardcoded series for pi
print(2 * (1 - 1 / 3.0 + 1 / 5.0 - 1 / 7.0 + 1 / 9.0 - 1 / 11.0 + 1 / 13.0 - 1 / 15.0
+ 1 / 17.0 - 1 / 19.0 + 1 / 21.0 - 1 / 23.0 + 1 / 25.0 - 1 / 27.0 + 1 / 29.0 - 1 / 31.0
/* slow convergence! */
+ 1 / 33.0 - 1 / 35.0 + 1 / 37.0 - 1 / 39.0 + 1 / 41.0 - 1 / 43.0 + 1 / 45.0 - 1 / 47.0
+ 1 / 49.0 - 1 / 51.0 + 1 / 53.0 - 1 / 55.0 + 1 / 57.0 - 1 / 59.0 + 1 / 61.0 - 1 / 63.0
+ 1 / 65.0 - 1 / 67.0 + 1 / 69.0 - 1 / 71.0 + 1 / 73.0 - 1 / 75.0 + 1 / 77.0 - 1 / 79.0
+ 1 / 81.0 - 1 / 83.0 + 1 / 85.0 - 1 / 87.0 + 1 / 89.0 - 1 / 91.0 + 1 / 93.0 - 1 / 95.0
+ 1 / 97.0 - 1 / 99.0 + 1 / 101.0 - 1 / 103.0 + 1 / 105.0 - 1 / 107.0 + 1 / 109.0 - 1 / 111.0
+ 1 / 113.0 - 1 / 115.0 + 1 / 117.0 - 1 / 119.0 + 1 / 121.0 - 1 / 123.0 + 1 / 125.0 - 1 / 127.0
+ 1 / 129.0 - 1 / 131.0 + 1 / 133.0 - 1 / 135.0 + 1 / 137.0 - 1 / 139.0 + 1 / 141.0 - 1 / 143.0
+ 1 / 145.0 - 1 / 147.0 + 1 / 149.0 - 1 / 151.0 + 1 / 153.0 - 1 / 155.0 + 1 / 157.0 - 1 / 159.0) * 2);

// Test continued fraction for pi
print(3.0 + 1.0 / (7 + 1 / (15 + 1 / (1 + 1 / (292 + 1 / (1 + 1 / (1 + 1 / (1 + 1 / (2 + 1 / (1 + 1 / (3 + 1 / (1 + 1 / (14))))) ))))))));

// Test Wallis product for pi
print(2 * (2.0 / 1.0 * 2.0 / 3.0 * 4.0 / 3.0 * 4.0 / 5.0 * 6.0 / 5.0 * 6.0 / 7.0));

// Test complicated formula
print(1 + 2 * 3 - 4 / 5 % 6 ** 7);
print(1 + 2 * 3 - 4.0 / 5 % 6 ** 7);

print(3141435 + 14814 * 1412 / 142.0 % 1341 ** 7 + (1 + 2 * 3 - 4 / 5 % 6 ** 7) - 23145);
