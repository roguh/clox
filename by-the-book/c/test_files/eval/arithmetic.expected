2
1
4
2
1
1
0
-2
4096
19
1
1
271825394
3.141592920353982
1
0.1415926500000002
8
1.618033988749895
2.718253968253968
3.129093141775721
3.142857142857143
2.925714285714285
7
6.2
3265602.408450704
